### PR TITLE
Allow nodes and edges to be removed

### PR DESCRIPTION
**Pull request by ruudk**
I need a way to remove nodes that were already added to the graph. With this change, it now takes care of the removal of the node and related edges.

### CODE REVIEW

File: relations.go [hunk 1]
```diff
@@ -42,6 +42,19 @@ func (relations *Relations) Add(parent string, child string) {
 	relations.ChildToParents[child][parent] = true
 }
 
+// Remove removes relation
+func (relations *Relations) Remove(parent string, child string) {
+	if _, ok := relations.ParentToChildren[parent]; !ok {
```
Comment (awalterschulze) [inline relations.go]: What about rather:

```go
if _, ok := relations.ParentToChildren[parent]; ok {
    delete(relations.ParentToChildren[parent], child)
}
```

And the same for the next if?
```

File: remove_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
+//Copyright 2013 GoGraphviz Authors
```
Comment (awalterschulze) [inline remove_test.go]: 2018 and you are are welcome to add yourself to the authors or contributors as you see appropriate

File: remove_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+package gographviz
+
+import (
+	"fmt"
+)
+
+func ExampleGraph_RemoveNode() {
```
Comment (awalterschulze) [inline remove_test.go]: Maybe we could add a few more tests?

Comment (ruudk) [reply remove_test.go]: You're right, will improve this with more tests.

File: remove_test.go (hunk 2 above)
Comment (ruudk) [reply remove_test.go]: Will do

File: relations.go (hunk 1 above)
Comment (ruudk) [reply relations.go]: Good one

**Comment by ruudk** [comment]
Maybe I can also try to remove subgraphs :D Will follow up on this, thanks!

**Comment by awalterschulze** [comment]
That would be great, but lets keep this pull request contained to this "single" feature.

Thank you so much for you work :)

File: remove_test.go (hunk 3 above)
Comment (ruudk) [reply remove_test.go]: Oops, I feel like a noob. I was under the impression that I added a test. But it was an example, with output verification. Changed it to be a test instead :) Hope it's good now.

Comment (awalterschulze) [reply remove_test.go]: I had no problem with the example, but I guess it is better as test.
And don't get me wrong this test covers a lot of cases, it is really cool.

I was wondering if you could add another test, that leaves some of the original nodes in the graph that had a connecting edge with the node that was removed.

Comment (ruudk) [reply remove_test.go]: Added more checkpoints and tests. Rebased the PR. Happy? :D

**Comment by awalterschulze** [comment]
I think just a gofmt run is missing to make the checks pass :)

File: remove_test.go (hunk 3 above)
Comment (awalterschulze) [reply remove_test.go]: Yes very :D

Comment (awalterschulze) [reply remove_test.go]: Thank you so much

**Comment by awalterschulze** [comment]
Thank you so much.
This is great :)
